includes/testing-library.sh

```diff
@@ -276,20 +313,27 @@ _stage_run_tests() {
     local test_selection
     local docroot=$(get_distribution_docroot)
     local composer_bin_dir=$(get_composer_bin_dir)
+    local project_type_directory=$(get_project_type_directory)
     local phpunit=${DRUPAL_TRAVIS_DRUPAL_INSTALLATION_DIRECTORY}/${composer_bin_dir}/phpunit
     local runtests=${docroot}/core/scripts/run-tests.sh
     local settings_file=${docroot}/sites/default/settings.php
-
-    if [ ${DRUPAL_TRAVIS_TEST_GROUP} ]; then
-       test_selection="--group ${DRUPAL_TRAVIS_TEST_GROUP}"
-    fi
+    local project_test_directory=${docroot}/${project_type_directory}/contrib/${DRUPAL_TRAVIS_PROJECT_NAME}
 
     case ${DRUPAL_TRAVIS_TEST_RUNNER} in
         "phpunit")
-            php ${phpunit} --verbose --debug --configuration ${docroot}/core ${test_selection} ${docroot}/modules/contrib/${DRUPAL_TRAVIS_PROJECT_NAME} || exit 1
+            if [ ${DRUPAL_TRAVIS_TEST_GROUP} ]; then
+               test_selection="--group ${DRUPAL_TRAVIS_TEST_GROUP}"
+            fi
+            php ${phpunit} --verbose --debug --configuration ${docroot}/core ${test_selection} ${project_test_directory} || exit 1
         ;;
         "run-tests")
-            php ${runtests} --php $(which php) --suppress-deprecations --verbose --color --url http://${DRUPAL_TRAVIS_HTTP_HOST}:${DRUPAL_TRAVIS_HTTP_PORT} ${DRUPAL_TRAVIS_TEST_GROUP} || exit 1
+            if [ ${DRUPAL_TRAVIS_TEST_GROUP} ]; then
+               test_selection="${DRUPAL_TRAVIS_TEST_GROUP}"
+            else
+               test_selection="--directory ${project_test_directory}"
+            fi
```

Comment:
Why is this mutual exclusive?

Reply:
Because I don't trust run-tests.sh to do anything sensible.

Reply:
And my mistrust is well placed. The options are exclusive.

Reply:
Wonderful :)